dipy/reconst/csdeconv.py

```diff
@@ -634,6 +641,119 @@ def auto_response(gtab, data, roi_center=None, roi_radius=10, fa_thr=0.7):
+        maximum number of iterations for calibration
+    convergence : float
+        convergence criterion, maximum relative change of SH coefficients
+
+    Returns
+    -------
+    response : ndarray
+        response function in SH coefficients
+
+    Notes
+    -----
+    In CSD there is an important pre-processing step: the estimation of the
+    fiber response function. Using an FA threshold is not a very robust method.
+    It is dependent on the dataset (non-informed used subjectivity), and still
+    depends on the diffusion tensor (FA and first eigenvector),
+    which has low accuracy at high b-value. This function recursively
+    calibrates the response function, for more information see [1].
+
+    References
+    ----------
+    .. [1] Tax, C.M.W., et al. NeuroImage 2014. Recursive calibration of
+           the fiber response function for spherical deconvolution of
+           diffusion MRI data.
+    """
+    S0 = 1
+    evals = fa_trace_to_lambdas(init_fa, init_trace)
+    response = (evals, S0)
+    sphere = get_sphere('symmetric724')
+
+    no_params = ((sh_order + 1) * (sh_order + 2)) / 2
+    response_p = np.ones(no_params)
+    if mask is None:
+        data = data.reshape(-1, data.shape[-1])
+#        data = data[np.ones(data.shape[0:(data.ndim-1)], dtype=bool)]
+    else:
+        data = data[mask]
+
+    m, n = sph_harm_ind_list(sh_order)
+    sh_mask = m != 0
+    where_dwi = lazy_index(~gtab.b0s_mask)
+
+    for num_it in range(1, iter):
+        r_sh_all = np.zeros(no_params)
+        csd_model = ConstrainedSphericalDeconvModel(gtab, response,
+                                                    None, sh_order)
+
+        csd_peaks = peaks_from_model(model=csd_model,
+                                     data=data,
+                                     sphere=sphere,
+                                     relative_peak_threshold=peak_thr,
+                                     min_separation_angle=25,
+                                     parallel=parallel)
+
+        dirs = csd_peaks.peak_dirs
+        vals = csd_peaks.peak_values
+        single_peak_mask = (vals[:, 1] / vals[:, 0]) < peak_thr
+        data = data[single_peak_mask]
+        dirs = dirs[single_peak_mask]
+
+        for num_vox in range(0, data.shape[0]):
+            rotmat = vec2vec_rotmat(dirs[num_vox, 0], np.array([0, 0, 1]))
+
+            rot_gradients = np.dot(rotmat, gtab.gradients.T).T
+
+            x, y, z = rot_gradients[where_dwi].T
+            r, theta, phi = cart2sphere(x, y, z)
+            # for the gradient sphere
+            B_dwi = real_sph_harm(m, n, theta[:, None], phi[:, None])
+            r_sh_all = r_sh_all + np.linalg.lstsq(B_dwi,
+                                                  data[num_vox, where_dwi])[0]
+
+        response = r_sh_all/data.shape[0]
+        response[sh_mask] = 0
+
+        change = abs((response_p[~sh_mask] - response[~sh_mask])/response_p[~sh_mask])
+        if change.all() < convergence:
```

Comment:
Hi, @ChantalTax  

Is `change.all() < convergence` what you intended or did you mean `all(change < convergence)`? As it is now, the result of `change.all()` (either True or False) is compared to `convergence` (0.001 by default).

Cheers,
Kesshi